### ISSUE TITLE
no system.cfg

### ISSUE THREAD

**Issue by Exarkuniv** [team-eternity/eternity#548]
i decided to try this again. now that i have a little better handle on how linux and whatnot works now
I can get it to comiple perfectly and i can even get my doom.wad to run.

i do get these to errors when i start eternity
```
Warning: Cannot read /home/pi/.config/eternity/user/system.cfg -- using built-in defaults
IWAD found: doom.wad
Ultimate DOOM version

Built on Dec 10 2021 at 17:03:01
M_LoadDefaults: Load system defaults.
Warning: Cannot read /home/pi/.config/eternity/user/doom/eternity.cfg -- using built-in defaults
```
i know they are just config files. but my question is when do they get made? or where do i find them. 

i will be making a install script for Retropie to install eternity and i want to make sure everything is working as it should. 
thanks for all your hard work on this

**Comment by Altazimuth** [comment]
The basic gist is that in the absence of them being in some sort of user-specific folder, the should be in `./user`, where `.` is the folder the Eternity executable is in. 

It's been a hot minute since I used my Pi or Linux, and I forget how exactly it works for it, so I'll need to dig a bit to try find whatever the code is that does the cfg copying (if there is any). At the very least, those warnings come from `M_LoadDefaultFile`.

**Comment by Altazimuth** [comment]
OK I found the priority list for where user is searched:
```
   // Priority:
   // 1. Command-line argument "-user"
   // 2. Environment variable "ETERNITYUSER"
   // 3. OS-specific home directory.
   // 4. /user under DoomExeDir
   // 5. /user under working directory
   // 6. basepath/../user
   // 7. use basepath itself.```
On Linux the OS-specific directories are `$XDG_CONFIG_HOME/eternity/user`, and `$HOME/.config/eternity/user`, the former having higher priority than the latter. The behaviour for these varies.

If the `$XDG_CONFIG_HOME` environment variable exists then the user directory will be set to that, otherwise if the `$HOME` environment variable exists the user directory will be set to that and `$HOME/.config` will be created.

After this, either of the environment variables are found then it will create the `eternity/user` folder and all the various user directories within it for each IWAD. When closing Eternity should save a file called `system.cfg` to the user folder, and `eternity.cfg` and `keys.csc` to the relevant IWAD folder nested within the user folder.

Does this issue repeat for you?

**Comment by Exarkuniv** [comment]
Yes it does make all the folders that you listed on my system. Im running RetorPie and they are on Buster.
```
/home/pi/.config/eternity/user
``` 
and in there i have doom, doom2, hacx, ect folders in there. but in each of those there is nothing.

i  can start the games with ```./eternity``` and the menu pops up for me to pick my WAD. and everything works. but even after i exit through the menu. no cfg file is made

**Comment by Exarkuniv** [comment]
ok i seem to have gotten it to work for me at lest. i had to go and remove the delete.me file from the user dir that was sitting in the build/eternity folder.

i will leave this open as i make the script incase something else pops up, but i think it should be good

**Comment by Exarkuniv** [comment]
what to thank you for the quick fix on the error i had earlier today.
now that i have it working i still have the issue that its not making a "eternity.cfg" or "keys.csc"
it did make a "system.cfg" so we are good there.

**Comment by Exarkuniv** [comment]
it is weird if i run ``sudo ./eternity doom2.wad`` i dont get that error when it starts but without sudo i do. 

i rather not use sudo since i just want what ever wad i pick just to start and not to use the launcher,
